plugins/janus_nosip.c

```diff
@@ -0,0 +1,2213 @@
+	srtp_crypto_policy_set_rtp_default(&(policy->rtp));
+	srtp_crypto_policy_set_rtcp_default(&(policy->rtcp));
+	if(suite == 32) {
+		srtp_crypto_policy_set_aes_cm_128_hmac_sha1_32(&(policy->rtp));
+		srtp_crypto_policy_set_aes_cm_128_hmac_sha1_32(&(policy->rtcp));
+	} else if(suite == 80) {
+		srtp_crypto_policy_set_aes_cm_128_hmac_sha1_80(&(policy->rtp));
+		srtp_crypto_policy_set_aes_cm_128_hmac_sha1_80(&(policy->rtcp));
+	}
+	policy->ssrc.type = ssrc_any_inbound;
+	policy->key = decoded;
+	policy->next = NULL;
+	/* Create SRTP context */
+	srtp_err_status_t res = srtp_create(video ? &session->media.video_srtp_in : &session->media.audio_srtp_in, policy);
+	if(res != srtp_err_status_ok) {
+		/* Something went wrong... */
+		JANUS_LOG(LOG_ERR, "Oops, error creating inbound SRTP session: %d (%s)\n", res, janus_nosip_get_srtp_error(res));
+		g_free(decoded);
+		policy->key = NULL;
+		return -2;
+	}
+	if((video && session->media.video_srtp_in) || (!video && session->media.audio_srtp_in)) {
+		JANUS_LOG(LOG_VERB, "%s inbound SRTP session created\n", video ? "Video" : "Audio");
+	}
+	return 0;
+}
+static void janus_nosip_srtp_cleanup(janus_nosip_session *session) {
+	if(session == NULL)
+		return;
+	session->media.autoack = TRUE;
+	session->media.require_srtp = FALSE;
+	session->media.has_srtp_local = FALSE;
+	session->media.has_srtp_remote = FALSE;
+	/* Audio */
+	if(session->media.audio_srtp_out)
+		srtp_dealloc(session->media.audio_srtp_out);
+	session->media.audio_srtp_out = NULL;
+	g_free(session->media.audio_local_policy.key);
+	session->media.audio_local_policy.key = NULL;
+	session->media.audio_srtp_suite_out = 0;
+	if(session->media.audio_srtp_in)
+		srtp_dealloc(session->media.audio_srtp_in);
+	session->media.audio_srtp_in = NULL;
+	g_free(session->media.audio_remote_policy.key);
+	session->media.audio_remote_policy.key = NULL;
+	session->media.audio_srtp_suite_in = 0;
+	/* Video */
+	if(session->media.video_srtp_out)
+		srtp_dealloc(session->media.video_srtp_out);
+	session->media.video_srtp_out = NULL;
+	g_free(session->media.video_local_policy.key);
+	session->media.video_local_policy.key = NULL;
+	session->media.video_srtp_suite_out = 0;
+	if(session->media.video_srtp_in)
+		srtp_dealloc(session->media.video_srtp_in);
+	session->media.video_srtp_in = NULL;
+	g_free(session->media.video_remote_policy.key);
+	session->media.video_remote_policy.key = NULL;
+	session->media.video_srtp_suite_in = 0;
+}
+
+
+/* SDP parsing and manipulation */
+void janus_nosip_sdp_process(janus_nosip_session *session, janus_sdp *sdp, gboolean answer, gboolean update, gboolean *changed);
+char *janus_nosip_sdp_manipulate(janus_nosip_session *session, janus_sdp *sdp, gboolean answer);
+/* Media */
+static int janus_nosip_allocate_local_ports(janus_nosip_session *session);
+static void *janus_nosip_relay_thread(void *data);
+
+
+/* Random string helper (in case user doesn't provide opaque info) */
```

Comment:
There is an info object users can provide in case they need some opaque identifier, but I'm questioning whether it's needed at all, considering you'll never bridge two sessions using this plugin directly... just in case it's needed, and the user doesn't provide any, then the random string kicks in.